### PR TITLE
Add release notes for v2.4.0 and update version

### PR DESCRIPTION
**Pull request by vshampor**

### CODE REVIEW

File: ReleaseNotes.md [hunk 1]
```diff
@@ -1,11 +1,42 @@
...
-The framework organized as a Python module that can be built and used as standalone or within
-samples distributed with the code.  The samples demonstrate the usage of compression methods on
-public models and datasets for three different use cases: Image Classification, Object Detection,
-and Semantic Segmentation.
+## New in Release 2.4.0
+Target version updates:
+- Bump target framework versions to PyTorch 1.13.1, TensorFlow 2.8.x, ONNX opset 13
+- Increased target HuggingFace transformers version for the integration patch to 4.23.1
+
+Features:
```
Comment (MaximProshin) [inline ReleaseNotes.md:8]: JPQD for several models has been added. I think it should be stated either as an experimental/preliminary support or for particular model.

Comment (vshampor) [reply ReleaseNotes.md:8]: Done

File: ReleaseNotes.md [hunk 2]
```diff
@@ -1,11 +1,42 @@
...
 
-## Introduction
-*Neural Network Compression Framework (NNCF)* is a toolset for Neural Networks model compression.
-The framework organized as a Python module that can be built and used as standalone or within
-samples distributed with the code.  The samples demonstrate the usage of compression methods on
-public models and datasets for three different use cases: Image Classification, Object Detection,
-and Semantic Segmentation.
+## New in Release 2.4.0
+Target version updates:
+- Bump target framework versions to PyTorch 1.13.1, TensorFlow 2.8.x, ONNX opset 13
```
Comment (kshpv) [inline ReleaseNotes.md:5]: ONNXRuntime to 1.13.1 ONNX to 1.12

File: ReleaseNotes.md [hunk 3]
```diff
@@ -1,11 +1,42 @@
...
+- Official release of the ONNX framework support.
+NNCF may now be used for post-training quantization (PTQ) on ONNX models.
+Added an example script demonstrating the ONNX post-training quantization on MobileNetV2.
+- Common post-training quantization API across the supported frameworks (PyTorch, TensorFlow and ONNX) via the `nncf.quantize(...)` function.
+The parameter set of the function is the same for all frameworks - actual framework-specific implementations are being dispatched based on the type of the model object argument.
+- (PyTorch) Joint Pruning, Quantization and Distillation for Transformers made possible via NNCF - 
+- (PyTorch, TensorFlow) Improved the adaptive compression training functionality to reduce effective training time.
+- (ONNX) Post-processing nodes are now automatically excluded from quantization.
+- (PyTorch - Experimental) Joint Pruning, Quantization and Distillation for Transformers enabled for certain models from HuggingFace `transformers` repo
+
```
Comment (kshpv) [inline ReleaseNotes.md:18]: (ONNX) Added FastBiasCorrection and BiasCorrection methods for Post-Training Quantization

File: ReleaseNotes.md [hunk 4]
```diff
@@ -1,11 +1,42 @@
...
+- "chunk" and "split" ops are correctly handled during pruning.
+- Linear layers may now be pruned by input and output independently.
+- Matmul-like operations and subsequent arithmetic operations are now treated as a fused pattern.
+- (PyTorch) Fixed a rare condition with accumulator overflow in CUDA quantization kernels, which led to CUDA runtime errors and NaN values appearing in quantized tensors and 
+- (PyTorch) `transformers` integration patch now allows to export to ONNX during training, and not only at the end of it.
+- (PyTorch) `torch.nn.utils.weight_norm` weights are now detected correctly.
+- (PyTorch) Exporting a model with sparsity or pruning no longer leads to weights in the original model object in-memory to be hard-set to 0.
+- (PyTorch, TensorFlow) Various bugs and issues with compression training were fixed.
+- (TensorFlow) Fixed an error with `"num_bn_adaptation_samples": 0` in config leading to a `TypeError` during quantization algo initialization.
+- (ONNX) Temporary model file is no longer saved on disk.
```
Comment (kshpv) [inline ReleaseNotes.md:32]: (ONNX) Fix incorrect quantizer setup for depthwise convolutions. Now they quantized in per-channel mode.

File: ReleaseNotes.md (hunk 3 above)
Comment (kshpv) [reply ReleaseNotes.md:18]: (ONNX) Added support of Preset

**Review by kshpv — CHANGES_REQUESTED** [review]

File: ReleaseNotes.md [hunk 5]
```diff
@@ -1,11 +1,42 @@
...
+Features:
+- Official release of the ONNX framework support.
+NNCF may now be used for post-training quantization (PTQ) on ONNX models.
+Added an example script demonstrating the ONNX post-training quantization on MobileNetV2.
+- Common post-training quantization API across the supported frameworks (PyTorch, TensorFlow and ONNX) via the `nncf.quantize(...)` function.
+The parameter set of the function is the same for all frameworks - actual framework-specific implementations are being dispatched based on the type of the model object argument.
+- (PyTorch) Joint Pruning, Quantization and Distillation for Transformers made possible via NNCF - 
+- (PyTorch, TensorFlow) Improved the adaptive compression training functionality to reduce effective training time.
+- (ONNX) Post-processing nodes are now automatically excluded from quantization.
+- (PyTorch - Experimental) Joint Pruning, Quantization and Distillation for Transformers enabled for certain models from HuggingFace `transformers` repo
```
Comment (ljaljushkin) [inline ReleaseNotes.md:17]: There are 2 bullet points about JPQD. I'd combine into one.

File: ReleaseNotes.md (hunk 3 above)
Comment (vshampor) [reply ReleaseNotes.md:18]: This is all part of PTQ for ONNX.

File: ReleaseNotes.md [hunk 6]
```diff
@@ -1,11 +1,42 @@
...
+## New in Release 2.4.0
+Target version updates:
+- Bump target framework versions to PyTorch 1.13.1, TensorFlow 2.8.x, ONNX opset 13
+- Increased target HuggingFace transformers version for the integration patch to 4.23.1
+
+Features:
+- Official release of the ONNX framework support.
+NNCF may now be used for post-training quantization (PTQ) on ONNX models.
+Added an example script demonstrating the ONNX post-training quantization on MobileNetV2.
+- Common post-training quantization API across the supported frameworks (PyTorch, TensorFlow and ONNX) via the `nncf.quantize(...)` function.
```
[the remark this answers is not in the document]
Comment (vshampor) [reply ReleaseNotes.md:12]: Done

File: ReleaseNotes.md (hunk 2 above)
Comment (vshampor) [reply ReleaseNotes.md]: Done

File: ReleaseNotes.md [hunk 7]
```diff
@@ -1,11 +1,43 @@
...
-and Semantic Segmentation.
+## New in Release 2.4.0
+Target version updates:
+- Bump target framework versions to PyTorch 1.13.1, TensorFlow 2.8.x, ONNX 1.12, ONNXRuntime 1.13.1
+- Increased target HuggingFace transformers version for the integration patch to 4.23.1
+
+Features:
+- Official release of the ONNX framework support.
+NNCF may now be used for post-training quantization (PTQ) on ONNX models.
+Added an example script demonstrating the ONNX post-training quantization on MobileNetV2.
```
Comment (kshpv) [inline ReleaseNotes.md:11]: Could you add the link to the examples?

**Review by AlexKoff88 — APPROVED** [review]

File: ReleaseNotes.md (hunk 5 above)
Comment (ljaljushkin) [reply ReleaseNotes.md:17]: The link to the new algorithm would be helpful here. Probably, it's related to other new features

**Review by ljaljushkin — CHANGES_REQUESTED** [review]

File: ReleaseNotes.md [hunk 8]
```diff
@@ -1,11 +1,43 @@
...
-public models and datasets for three different use cases: Image Classification, Object Detection,
-and Semantic Segmentation.
+## New in Release 2.4.0
+Target version updates:
+- Bump target framework versions to PyTorch 1.13.1, TensorFlow 2.8.x, ONNX 1.12, ONNXRuntime 1.13.1
+- Increased target HuggingFace transformers version for the integration patch to 4.23.1
+
+Features:
+- Official release of the ONNX framework support.
+NNCF may now be used for post-training quantization (PTQ) on ONNX models.
```
Comment (kshpv) [inline ReleaseNotes.md:10]: I suggest to add the links to Readme of PTQ or small description about this feature

**Review by kshpv — CHANGES_REQUESTED** [review]

File: ReleaseNotes.md (hunk 7 above)
Comment (vshampor) [reply ReleaseNotes.md:11]: Done

**Review by MaximProshin — APPROVED** [review]

File: ReleaseNotes.md (hunk 5 above)
Comment (vshampor) [reply ReleaseNotes.md]: Done

File: ReleaseNotes.md [hunk 9]
```diff
@@ -1,11 +1,43 @@
...
+- Linear layers may now be pruned by input and output independently.
+- Matmul-like operations and subsequent arithmetic operations are now treated as a fused pattern.
+- (PyTorch) Fixed a rare condition with accumulator overflow in CUDA quantization kernels, which led to CUDA runtime errors and NaN values appearing in quantized tensors and 
+- (PyTorch) `transformers` integration patch now allows to export to ONNX during training, and not only at the end of it.
+- (PyTorch) `torch.nn.utils.weight_norm` weights are now detected correctly.
+- (PyTorch) Exporting a model with sparsity or pruning no longer leads to weights in the original model object in-memory to be hard-set to 0.
+- (PyTorch, TensorFlow) Various bugs and issues with compression training were fixed.
+- (TensorFlow) Fixed an error with `"num_bn_adaptation_samples": 0` in config leading to a `TypeError` during quantization algo initialization.
+- (ONNX) Temporary model file is no longer saved on disk.
+- (ONNX) Depthwise convolutions are now quantizable in per-channel mode.
```
Comment (ljaljushkin) [inline ReleaseNotes.md:33]: (Pytorch - Experimental) improved automatic search of blocks to skip within the NAS algorithm – overlapping blocks are correctly filtered.

Comment (vshampor) [reply ReleaseNotes.md:34]: Done

File: ReleaseNotes.md [hunk 10]
```diff
@@ -1,11 +1,44 @@
...
+
+Breaking changes:
+- Fused patterns will be excluded from quantization via `ignored_scopes` only if the top-most node in data flow order matches against `ignored_scopes`
+- NNCF config's `"ignored_scopes"` and `"target_scopes"` are now strictly checked to be matching against at least one node in the model graph instead of silently ignoring the unmatched entries.
+- Calling `setup.py` directly to install NNCF is deprecated and no longer guaranteed to work.
+- Importing NNCF logger as `from nncf.common.utils.logger import logger as nncf_logger` is deprecated - use `from nncf import nncf_logger` instead.
+- `pruning_rate` is renamed to `pruning_level` in pruning compression controllers.
```
Comment (kshpv) [inline ReleaseNotes.md:41]: Removed CompressionBuilder. Excluded examples of NNCF for ONNX with CompressionBuilder API

File: ReleaseNotes.md [hunk 11]
```diff
@@ -1,11 +1,44 @@
...
+- (PyTorch) Fixed a rare condition with accumulator overflow in CUDA quantization kernels, which led to CUDA runtime errors and NaN values appearing in quantized tensors and 
+- (PyTorch) `transformers` integration patch now allows to export to ONNX during training, and not only at the end of it.
+- (PyTorch) `torch.nn.utils.weight_norm` weights are now detected correctly.
+- (PyTorch) Exporting a model with sparsity or pruning no longer leads to weights in the original model object in-memory to be hard-set to 0.
+- (PyTorch - Experimental) improved automatic search of blocks to skip within the NAS algorithm – overlapping blocks are correctly filtered.
+- (PyTorch, TensorFlow) Various bugs and issues with compression training were fixed.
+- (TensorFlow) Fixed an error with `"num_bn_adaptation_samples": 0` in config leading to a `TypeError` during quantization algo initialization.
+- (ONNX) Temporary model file is no longer saved on disk.
+- (ONNX) Depthwise convolutions are now quantizable in per-channel mode.
+
```
Comment (kshpv) [inline ReleaseNotes.md:35]: (ONNX) Improved the working time of PTQ by optimizing the call numbers of shape_inference()

**Review by kshpv — CHANGES_REQUESTED** [review]

File: ReleaseNotes.md (hunk 10 above)
Comment (vshampor) [reply ReleaseNotes.md:41]: It was never exposed, so I don't think it has to be in the release notes.

File: ReleaseNotes.md (hunk 11 above)
Comment (vshampor) [reply ReleaseNotes.md:35]: Done

File: ReleaseNotes.md (hunk 10 above)
Comment (kshpv) [reply ReleaseNotes.md:42]: We used this API in our ONNX examples, which some users got to use.

File: ReleaseNotes.md (hunk 8 above)
Comment (kshpv) [reply ReleaseNotes.md:10]: I will add this in https://github.com/openvinotoolkit/nncf/pull/1505

File: ReleaseNotes.md (hunk 10 above)
Comment (vshampor) [reply ReleaseNotes.md:42]: Done

**Review by kshpv — APPROVED** [review]

File: ReleaseNotes.md [hunk 12]
```diff
@@ -1,11 +1,46 @@
...
+- Bump target framework versions to PyTorch 1.13.1, TensorFlow 2.8.x, ONNX 1.12, ONNXRuntime 1.13.1
+- Increased target HuggingFace transformers version for the integration patch to 4.23.1
+
+Features:
+- Official release of the ONNX framework support.
+NNCF may now be used for post-training quantization (PTQ) on ONNX models.
+Added an [example script](examples/post_training_quantization/onnx/mobilenet_v2) demonstrating the ONNX post-training quantization on MobileNetV2.
+- Common post-training quantization API across the supported framework model formats (PyTorch, TensorFlow, ONNX, OpenVINO IR) via the `nncf.quantize(...)` function.
+The parameter set of the function is the same for all frameworks - actual framework-specific implementations are being dispatched based on the type of the model object argument.
+- (PyTorch, TensorFlow) Improved the adaptive compression training functionality to reduce effective training time.
```
[the remark this answers is not in the document]
Comment (vshampor) [reply ReleaseNotes.md:17]: Done

**Review by alexsu52 — APPROVED** [review]

**Pull request merged**